src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -343,9 +413,25 @@ void setNotifier(Notifier* n) {
 */
 DBClientBase* getMongoConnection(void)
 {
-    return connection;
+#ifdef UNIT_TEST
+  return connection;
+#else
+  return mongoPoolConnectionGet();
+#endif
 }
 
+DBClientBase* mongoInitialConnectionGetForUnitTest(void)
+{
+  return mongoPoolConnectionGet();
+}
+
+void releaseMongoConnection(DBClientBase* connection)
```

Comment:
This function seems to be a direct pass-through. Why keep it?

Or maybe, for symmetry, it should be:

```
void releaseMongoConnection(DBClientBase* connection)
#ifndef UNIT_TEST
 mongoPoolConnectionRelease(connection)
#endif
 }
```
Although I know that mongoPoolConnectionRelease() doesn't need to be "protected" in the case of unit test code, it would help to read this code.